### ISSUE TITLE
TPU Training Error: File system scheme '[local]' not implemented

### ISSUE THREAD

**Issue by acsboy** [tensorflow/models#8265]
Hi. I am trying to execute training using Google TPU.

It says: Error recorded from training_loop: File system scheme '[local]' not implemented (file: '/tmp/tmpUUrctt/model.ckpt-0_temp_3546763fe0ab4b32a5353eb8f190192c')

I googled and found that it is a common error and their recommended solution is: All input files and the model directory must use a cloud storage bucket path (gs://bucket-name/...), and this bucket must be accessible from the TPU server.

I have doubled check and confirmed the following:
- Cloud storage API is already enabled.
- GCP Project has been made the "Storage Legacy Bucket Owner"

Can I be missing anything? Any help is much appreciated. Thank you.

**Comment by saberkun** [comment]
Hi, which models are you using?

Here are some tutorials: https://cloud.google.com/tpu/docs/tutorials/resnet-2.x

Adding a few cloud tpu team members to check out here. 
@allenwang28 @gagika

**Comment by allenwang28** [comment]
It looks like model_dir is set to a local directory in /tmp/, can you double check to make sure that it is a path to a GCS bucket?

**Comment by acsboy** [comment]
@saberkun I am using ssd_mobilenet_v1_0.75 model.

@allenwang28 The training procedure has created a new directory in my bucket "model_dir" before error-ing. I assume that means the path is already correct.

**Comment by allenwang28** [comment]
It could be possible that it's creating a directory within data_dir which is a GCS path. Both data_dir and model_dir should be set to GCS buckets

**Comment by sayakpaul** [comment]
Hi. 

I was able to execute on Cloud v3 TPUs using local files.

**Comment by hanshengchiu** [comment]
Does this GCS requirement apply to tensorboard's log dir too? i.e. tf.summary.create_file_writer(logdir...)

**Comment by allenwang28** [comment]
Yes, for Cloud TPU usage, `model_dir` (which is the parent directory for most models' `logdir`) must be a GCS bucket. As a sanity check, I've run an experiment where I replaced `logdir` (where `data_dir`, `model_dir` were GCS buckets) with a local directory which failed with this same error.

**Comment by soltaniehha** [comment]
I am getting this error in Colab mounting Drive as my storage. All of my training data and checkpoints are in Drive. Is there a way to use TPU in Colab with this setup?

**Comment by sayakpaul** [comment]
You would need to have your data stored inside a GCS bucket and the Cloud TPUs should have access to that. 

You are probably better off using Kaggle TPUs here.

**Comment by soltaniehha** [comment]
Thank you for your response. The big challenge with accessing GCS data is the storage egress charge that I have been acquiring since Colab sometimes randomly assigns a VM in EU or APAC. This makes Colab TPUs very impractical for any non-public dataset. I would like to take advantage of TPUs with my Colab pro account but I guess it's not possible at this moment. The ability to read local tfrecords should be something standard IMHO. I hope this feature can be added soon.

**Comment by BioWar** [comment]
@soltaniehha How do you check the region of your Colab VM?

**Comment by soltaniehha** [comment]
@BioWar You can simply look up the IP address of your VM (something like `!curl ifconfig.me` in your Colab cell) and then look up which geolocation that IP belongs to. Probably there are easier ways to do it but this worked for me.

**Comment by soltaniehha** [comment]
@oscarjose9423 - this seems to work fine. I ended up sticking with GPUs with Colab.
